expect/_matchers.ts

```diff
@@ -26,14 +26,37 @@ export function toBe(context: MatcherContext, expect: unknown): MatchResult {
   }
 }
 
+// deno-lint-ignore no-explicit-any
+function filterUndefined(obj: any): any {
+  if (typeof obj !== "object") return obj;
+  if (obj instanceof Date) return obj;
+  if (obj instanceof RegExp) return obj;
+  if (ArrayBuffer.isView(obj)) return obj;
+  if (obj === null) return obj;
+  if ("size" in obj) return obj;
+  if (Symbol.iterator in obj) return obj;
+  if (Array.isArray(obj)) return obj.map(filterUndefined);
```

Comment:
Nit: Perhaps this logic could be factored out into its own `isPlainObject()` just so its purpose is made clear.